presto-tests/src/test/java/com/facebook/presto/tests/TestLocalQueries.java

```diff
@@ -58,4 +59,20 @@ public static LocalQueryRunner createLocalQueryRunner()
 
         return localQueryRunner;
     }
+
+    @Test
+    public void testInequalityJoinWithDate()
+            throws Exception
+    {
+        assertQuery(
+                "select o.orderkey, o.orderdate, l.shipdate from orders o, lineitem l where l.orderkey=o.orderkey and l.shipdate < o.orderdate + INTERVAL '10' DAY",
+                "select o.orderkey, o.orderdate, l.shipdate from orders o, lineitem l where l.orderkey=o.orderkey and l.shipdate < DATEADD('DAY', 10, o.orderdate)");
```

Comment:
Generally, I think these tests are not enough. It is similar with the current problem related to `Exchanges`.
The query will pass regardless you other commits are merged or not.

I am sure there are already some tests form inequality join.